plotly/tools.py

```diff
@@ -2093,14 +2088,717 @@ def get_streamline_arrows(self):
+                                                        increase_open,
+                                                        increase_dif,
+                                                        stick_increase_y,
+                                                        stick_increase_x,
+                                                        **kwargs)
+
+            data = [hidden_bar_incr, candle_bar_incr, candle_line_incr]
+
+        elif direction is 'decreasing':
+            (hidden_bar_decr,
+             candle_bar_decr,
+             candle_line_decr) = make_decreasing_candle(decrease_x,
+                                                        decrease_close,
+                                                        decrease_dif,
+                                                        stick_decrease_y,
+                                                        stick_decrease_x,
+                                                        **kwargs)
+            data = [hidden_bar_decr, candle_bar_decr, candle_line_decr]
+
+        elif direction is 'both':
+            (hidden_bar_incr,
+             candle_bar_incr,
+             candle_line_incr) = make_increasing_candle(increase_x,
+                                                        increase_open,
+                                                        increase_dif,
+                                                        stick_increase_y,
+                                                        stick_increase_x,
+                                                        **kwargs)
+            (hidden_bar_decr,
+             candle_bar_decr,
+             candle_line_decr) = make_decreasing_candle(decrease_x,
+                                                        decrease_close,
+                                                        decrease_dif,
+                                                        stick_decrease_y,
+                                                        stick_decrease_x,
+                                                        **kwargs)
+            data = [hidden_bar_incr, candle_bar_incr, candle_line_incr,
+                    hidden_bar_decr, candle_bar_decr, candle_line_decr]
+
+        return dict(data=data, layout=layout)
+
+
+class _OHLC(dict):
+    """
+    Refer to TraceFactory.create_ohlc_increase() for docstring.
+    """
+    def __init__(self, open, high, low, close, dates, **kwargs):
+        self.open = open
+        self.high = high
+        self.low = low
+        self.close = close
+        self.empty = [None] * len(open)
+        if dates:
+            self.dates = dates
+        else:
+            self.dates = None
+        self.all_x = []
+        self.all_y = []
+        self.increase_x = []
+        self.increase_y = []
+        self.decrease_x = []
+        self.decrease_y = []
+        self.get_all_xy()
+        self.separate_increase_decrease()
+        self.get_increase()
+        self.get_decrease()
```

Comment:
What's the point of calling `get_increase()` and `get_decrease()` here? They don't seem to change the state and the return values aren't being caught.